Sources/ArcGISToolkit/Components/OfflineMapAreasView/PreplannedMapModel.swift

```diff
@@ -0,0 +1,128 @@
+// Copyright 2024 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import SwiftUI
+import ArcGIS
+
+/// An object that encapsulates state about a preplanned map.
+@MainActor
+class PreplannedMapModel: ObservableObject, Identifiable {
+    /// The preplanned map area.
+    let preplannedMapArea: any PreplannedMapAreaProtocol
```

Comment:
For some reason my compiler doesn't give a warning when I omit the `any`. Since it is a Swift 6 requirement, it's good to have it.